dev/NavigationView/NavigationView_ApiTests/NavigationViewTests.cs

```diff
@@ -759,5 +759,70 @@ public void VerifyNavigationViewItemInFooterDoesNotCrash()
                 Verify.IsTrue(true);
             });
         }
+
+        [TestMethod]
+        public void VerifyExpandCollapseChevronVisibility()
```

Comment:
* For now we check if the chevron is collapsed when the bound data collection is empty. We should probably also check in this test if the chevron visibility is updated accordingly when the NavigationViewItem.MenuItemsSource is set to `null`. (I.e. you could add an item again to the bound collection after our final visibility check after clearing the collection, then set `parentItem.MenuItemsSource = null;` and verify again if the chevron is collapsed.

* In this test method, right now we only use the `NavigationViewItem.MenuItemsSource` API to verify the correctness of the chevron visibilty. We could also add testing if the chevron visibility is updated correctly when using the `NavigationViewItem.MenuItems` API instead. Testing with this API would simply add/remove items to/from MenuItems (as MenuItems cannot be set to `null`) and check the chevron visibility accordingly. These checks could be added inside this method below our tests for the `MenuItemsSource` API (for example after clearing the bound collection or setting MenuItemsSource to null) so that we start testing using the MenuItems API with the chevron in collapsed state.

Reply:
Are there no other tests that would verify the other branches of the `HasChildren` function? I thought that some other tests verified this implicitly.

Reply:
@chingucoding I haven't found any other test (API/Interaction test) which consumes the `HasChildren` API or tests it explicitly. That raises a good point though. This PR after all fixed two kinds of issues - a bug in the `HasChildren()` API and an issue that the chevron visibility was not being updated after children were added/removed. As such, I think we should also add a test here verifying the correctness of the HasChildren API. The test code would be quite similar in both the "VerifyHasChildren()" test method and the `VerifyExpandCollapseChevronVisibility()` test method yet I think there's a case to make to have two separate test mthods here (chevron is part of the UI and could change over time whereas HasChildren is an API and should be quite stable).

Reply:
It seems that are no tests that test collection changes in the MenuItems API. Would be best if we add those here or if we create a tracking issue to do that later.

Reply:
I am adding the tests for the MenuItems API now. Will push it once it passes the tests.

Reply:
All tests have passed even the tests for the MenuItems API.

![image](https://user-images.githubusercontent.com/16188252/86136727-01650780-bb1f-11ea-8173-bdcd86a54b90.png)